### DISCUSSION TITLE
Unexpected token 'A', "An error o"... is not valid JSON

### DISCUSSION THREAD

**Discussion by BishalN** [trpc/trpc#3640] [General]
I am getting this strange error on production but can't replicate it in my dev environment
Production is hosted on vercel
AnyHelp would be highly appreciated 
Thanks

**Comment by BishalN** [comment]
```
 Uncaught (in promise) TRPCClientError: Unexpected token 'A', "An error o"... is not valid JSON
```

**Comment by stephenasuncionDEV** [comment]
Try returning a parsed JSON.

```ts
return JSON.parse(JSON.stringify(jsonObj));
```

**Comment by MrEmanuel** [comment]
I encountered this problem when I tried to do JSON.parse on a string without it containing json data. It was just normal text.   
```JSON.parse(error.message)```

**Comment by kac487** [comment]
I also encountered this issue when using TRPC in my web app running on vercel. I spent waay too long trying to figure out why I was only able to re-produce this error after deploying the app. 

It turns out vercel was throwing a FUNCTION_INVOCATION_TIMEOUT and rug-pulling my trpc procedure before it could finish. I had to upgrade my vercel account from a hobby account to pro subscription to extend the timeout from 10 seconds to something longer.

Hope this saves someone some time.

**Reply by Saralhossain** [reply]
This ans seems logical. I'm facing "SyntaxError: Unexpected token 'A', "An error o"... is not valid JSON" even tho it's working on my dev but when I deployed to UAT I'm facing this error.

**Reply by ForeignBrazilian** [reply]
this was actual problem for me thank you!

**Comment by Nick-Lucas** [comment]
Anyone who sees this type of error, check your network debugger and inspect the actual response from the server.

Chances are it's infrastructure related, usually some HTML being passed back because the tRPC route doesn't even get reached.

**Comment by volcinschi** [comment]
This happens only in production because your procedure is timing out most likely

**Comment by MrAlaminH** [comment]
This is because of the Vercel  TIMEOUT FUNCTION issue if you are using Vercel free tire you can solve it by adding a vercel.json in the root folder. this code gonna increase the time-out durations from default 10s to 60s if you need more durations for your work then you must have to upgrade the Vercel free to their premium plan. 

```js
// vercel.json

{
  "functions": {
    "app/api/**/route.ts": {
      "maxDuration": 60
    }
  }
}
```

**Reply by LyghtCode** [reply]
For page router

`{
  "functions": {
    "pages/api/**/*": {
      "maxDuration": 5 // All functions can run for a maximum of 5 seconds
    }
  }
}`
